### PR TITLE
Add NodeMcu ESP32-C2

### PR DESCRIPTION
**Pull request by bill88t**
Yet another aliexpress board that is readily available.
It's the first decent C2 board I found, so I picked one up.

Works fine, connected to a bunch of wifi networks.

Notes:
 - The two sets of uart pins are internally connected, even if they are labeled differently.
 - There is no status indicator at all.
 
Pending creator id stuff.

### CODE REVIEW

File: ports/espressif/boards/nodemcu_esp32c2/board.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
```
Comment (tannewt) [inline ports/espressif/boards/nodemcu_esp32c2/board.c:4]: Headers have been updated to SPDX. Is this based on an old branch?

Comment (bill88t) [reply ports/espressif/boards/nodemcu_esp32c2/board.c]: `espressif_esp8684_devkitc_02_n4` which I used as a base, wasn't updated apparently.
Updated.

Comment (tannewt) [reply ports/espressif/boards/nodemcu_esp32c2/board.c]: Ok, thanks! I'll up date it too.

**Review by tannewt — APPROVED** [review]
Thanks for updating the other board too!

**Pull request merged**